PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalClient.kt

```diff
@@ -152,10 +155,20 @@ class PayPalClient internal constructor(
 
         return BrowserSwitchOptions()
             .requestCode(BraintreeRequestCodes.PAYPAL.code)
-            .appLinkUri(merchantRepository.appLinkReturnUri)
             .url(Uri.parse(paymentAuthRequest.approvalUrl))
             .launchAsNewTask(braintreeClient.launchesBrowserSwitchAsNewTask())
             .metadata(metadata)
+            .apply {
+                when (getReturnLinkTypeUseCase()) {
+                    GetReturnLinkTypeUseCase.ReturnLinkType.APP_LINK -> {
+                        appLinkUri(merchantRepository.appLinkReturnUri)
+                    }
+
+                    GetReturnLinkTypeUseCase.ReturnLinkType.DEEP_LINK -> {
+                        returnUrlScheme(merchantRepository.returnUrlScheme)
```

Comment:
Do we need to check that returnUrlScheme isn't null?

Reply:
That's a good question. I think we should check if returnUrlScheme is non-null. However, how should we handle the null case?

If the buyer has the `Open supported links` unchecked, and the merchant passes a null `returnUrlScheme` the deep link would fail. Maybe we should throw an error before the app switch/CCT is launched and fail fast?

Reply:
Yes I think fail fast and let the merchant decide if they want to tell the user to turn on that setting or how they want to handle it if they don't want to support deep links

Reply:
We're actually defaulting the returnUrlScheme to the merchant's package name appended with `.braintree`, if a null value is passed in.

https://github.com/braintree/braintree_android/blob/bafeda710b4c65780395ef2f1ca872254d648e9c/BraintreeCore/src/main/java/com/braintreepayments/api/core/BraintreeClient.kt#L54

So the `returnUrlScheme` in `MerchantRepository` is non null. I believe this was the existing deep link behavior.

Reply:
Yep, it would fail if they do not define the intent filter. Hmm I guess having it more explicit would be better for the deep link fallback. I'm thinking we may need to have a new field that stores the deep link fallback URL scheme if the other modules are still relying on the existing URL scheme with this default value.

Reply:
I just verified that we're using the existing `returnUrlScheme` with the default value for the other modules. I'll add a new param to the MerchantRepository for holding the fallback URL scheme.

Reply:
I also like the explicit opt-in to use a deep link fallback. I can see some merchants only wanting to support deep links for the increased security.

Another option could be to reuse the app link and have the merchant pass in a boolean value if they want to enable the deep link fallback, but they'd still need to add the correct intent filter in their manifest. So it might be clearer to have them pass in that URL scheme they define in their manifest.

Reply:
go ahead :)